test/Datadog.Trace.Tests/TracerTests.cs

```diff
@@ -18,6 +23,14 @@ public class TracerTests
     {
         private readonly Tracer _tracer;
 
+        static TracerTests()
+        {
+#if NET452
+            LifetimeServices.LeaseTime = TimeSpan.FromMilliseconds(100);
+            LifetimeServices.LeaseManagerPollTime = TimeSpan.FromMilliseconds(10);
```

Comment:
10ms seems awfully low for anything that has "poll" in the name. Unfortunately I'm unfamiliar with this API, but it seems to command a timer that acquires a lock before doing a bunch of stuff: https://referencesource.microsoft.com/#mscorlib/system/runtime/remoting/leasemanager.cs,be02ebedeb207a56

I'm worried about the impact. Is there a reason to change the default value? (10 seconds) If it's to make sure the memory is reclaimed eagerly then 1 second seems a more reasonable value to me (since that the delay before flushing the spans)

Reply:
This is only in the test class, not the product code. The main code change just makes sure that every time a Sponsor is asked if they want to renew the lease, we renew the lease until we remove the object from the Logical Call Context.

Reply:
Right, sorry